controllers/etcd_copy_backups_controller.go

```diff
@@ -0,0 +1,446 @@
+	imageVector, err := imagevector.ReadGlobalImageVectorWithEnvOverride(getImageYAMLPath())
+	if err != nil {
+		return nil, err
+	}
+	r.ImageVector = imageVector
+	return r, nil
+}
+
+// Reconcile reconciles the EtcdCopyBackupsTask.
+func (r *EtcdCopyBackupsTaskReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	copyTask := &druidv1alpha1.EtcdCopyBackupsTask{}
+	if err := r.Get(ctx, req.NamespacedName, copyTask); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Object not found, return. Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	if !copyTask.DeletionTimestamp.IsZero() {
+		return r.delete(ctx, copyTask)
+	}
+	return r.reconcile(ctx, copyTask)
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) reconcile(ctx context.Context, copyTask *druidv1alpha1.EtcdCopyBackupsTask) (ctrl.Result, error) {
+	logger := r.logger.WithValues("etcd", kutil.Key(copyTask.Namespace, copyTask.Name).String(), "operation", "reconcile")
+	logger.Info("Starting operation")
+
+	if err := r.addFinalizerToObject(ctx, copyTask, logger); err != nil {
+		return ctrl.Result{
+			Requeue: true,
+		}, err
+	}
+
+	if err := r.removeOperationAnnotation(ctx, logger, copyTask); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{
+			Requeue: true,
+		}, err
+	}
+
+	if !copyTask.Status.TaskDeployed {
+		logger.Info("Deploying ETCD copy backups job")
+		if err := r.checkForDuplicateOperation(ctx, copyTask); err != nil {
+			if statusError := r.updateEtcdCopyTaskStatusLastError(ctx, copyTask, err); statusError != nil {
+				return ctrl.Result{}, fmt.Errorf("%s; %s", err, statusError)
+			}
+			return ctrl.Result{}, err
+		}
+
+		values, err := r.getValuesMap(copyTask)
+		var job *batchv1.Job
+		if err != nil {
+			return ctrl.Result{
+				Requeue: true,
+			}, err
+		}
+		if job, err = r.runCopyJob(ctx, copyTask, values, logger); err != nil {
+			if statusError := r.updateEtcdCopyTaskStatusLastError(ctx, copyTask, err); statusError != nil {
+				return ctrl.Result{}, fmt.Errorf("%s; %s", err, statusError)
+			}
+			return ctrl.Result{
+				Requeue: true,
+			}, err
+		}
+		if err := r.addFinalizerToObject(ctx, job, logger); err != nil {
+			return ctrl.Result{
+				Requeue: true,
+			}, err
+		}
+		if err := r.updateEtcdCopyTaskStatusWithTaskDescription(ctx, copyTask, job.Name, reflect.TypeOf(job).Elem().Name(), batchv1.SchemeGroupVersion.String()); err != nil {
+			return ctrl.Result{
+				Requeue: true,
+			}, err
+		}
+	}
+
+	if err := r.updateEtcdCopyBackupsTaskStatus(ctx, copyTask); err != nil {
+		return ctrl.Result{
+			Requeue: true,
+		}, err
+	}
+
+	return ctrl.Result{
+		Requeue:      true,
+		RequeueAfter: 30 * time.Second,
+	}, nil
```

Comment:
Why always requeue? I think the controller should only requeue until the job has completed, either successfully or with an error.